src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php

```diff
@@ -138,6 +138,19 @@ public function registerProvider(ExpressionFunctionProviderInterface $provider)
     protected function registerFunctions()
     {
         $this->addFunction(ExpressionFunction::fromPhp('constant'));
+
+        $this->addFunction(new ExpressionFunction('enum',
+            static fn ($str): string => sprintf("(\$v = \constant(%1\$s)) instanceof \UnitEnum ? \$v : throw new \TypeError('The string %1\$s is not the name of a valid enum case.')", $str),
```

Comment:
be careful. Here `$str` will be the compiled PHP expression, which might not be a literal string. The error message might be confusing for such cases

Reply:
`static fn ($str): string => sprintf("(\constant($str = (%1\$s))) instanceof \UnitEnum ? \constant(\$v) : throw new \TypeError(\sprintf('The string "%s" is not the name of a valid enum case.', $str))", $str),`

Reply:
Thank you both, updated!